Sources/StreamChat/Controllers/SearchControllers/MessageSearchController/MessageSearchController_Tests.swift

```diff
@@ -0,0 +1,649 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+@testable import StreamChat
+@testable import StreamChatTestTools
+import XCTest
+
+class MessageSearchController_Tests: XCTestCase {
+    fileprivate var env: TestEnvironment!
+    
+    var client: ChatClient!
+    
+    var query: MessageSearchQuery!
+    var controller: ChatMessageSearchController!
+    var controllerCallbackQueueID: UUID!
+    /// Workaround for unwrapping **controllerCallbackQueueID!** in each closure that captures it
+    private var callbackQueueID: UUID { controllerCallbackQueueID }
+    
+    override func setUpWithError() throws {
+        super.setUp()
+        
+        env = TestEnvironment()
+        client = ChatClient.mock
+        query = .init(
+            channelFilter: .exists(.cid),
+            messageFilter: .queryText("")
+        )
+        controller = ChatMessageSearchController(client: client, environment: env.environment)
+        controllerCallbackQueueID = UUID()
+        controller.callbackQueue = .testQueue(withId: controllerCallbackQueueID)
+        // Message search requires a current user
+        client.currentUserId = .unique
+    }
+    
+    override func tearDown() {
+        controllerCallbackQueueID = nil
+        
+        env.messageUpdater?.cleanUp()
+        
+        AssertAsync {
+            Assert.canBeReleased(&controller)
+            Assert.canBeReleased(&client)
+            Assert.canBeReleased(&env)
+        }
```

Comment:
I would be careful in using this. When I fixed some memory leaks in the previous PR, this `Assert.canBeReleased()` function was not actually catching the memory leak.

Reply:
Here, it works as expected